IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -147,15 +147,14 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
             MSIDTokenResult *tokenResult = [self resultWithAccessToken:accessToken
                                                           refreshToken:refreshableToken
                                                                  error:&resultError];
-
-            if (resultError)
+            
+            if (tokenResult)
             {
-                completionBlock(nil, resultError);
+                completionBlock(tokenResult, nil);
                 return;
             }
-
-            completionBlock(tokenResult, nil);
-            return;
+            
+            MSID_LOG_WARN_PII(nil, @"Couldn't create result for cached access token, error %@. Try to recover...", resultError);
         }
 
         if (accessToken && accessToken.isExtendedLifetimeValid)
```

Comment:
When id token/account is missing, should we stop returning it as extended-life token the same way as above?

Reply:
It's handled already, the logic that returns extended life token would get nil token result.
If server is unavailable, it will return error to developer:

```
if (serverUnavailable && self.requestParameters.extendedLifetimeEnabled && self.extendedLifetimeAccessToken)
            {
                NSTimeInterval expiresIn = [self.extendedLifetimeAccessToken.extendedExpireTime timeIntervalSinceNow];
                MSID_LOG_INFO(self.requestParameters, @"Server unavailable, using long-lived access token, which expires in %f", expiresIn);
                NSError *cacheError = nil;
                MSIDTokenResult *tokenResult = [self resultWithAccessToken:self.extendedLifetimeAccessToken
                                                              refreshToken:refreshToken
                                                                     error:&cacheError];
                tokenResult.extendedLifeTimeToken = YES;

                completionBlock(tokenResult, cacheError);
                return;
            }
```

But perhaps better error would be good. let me take a look.